lib/streamlit/elements/lib/column_config_utils.py

```diff
@@ -382,8 +382,8 @@ def determine_dataframe_schema(
 
 
 # A mapping of column names/IDs to column configs.
-ColumnConfigMapping: TypeAlias = Dict[Union[IndexIdentifierType, str], ColumnConfig]
-ColumnConfigMappingInput: TypeAlias = Dict[
+ColumnConfigMapping: TypeAlias = Mapping[Union[IndexIdentifierType, str], ColumnConfig]
```

Comment:
I think the `ColumnConfigMapping` needs to be either a `MutableMapping` or changed back to `Dict` since we use this to modify the structure internally. For `ColumnConfigMappingInput` I think it is fine to be just a mapping.

Reply:
yeah, makes sense.